source/ceylon/net/http/server/internal/RequestImpl.ceylon

```diff
@@ -159,33 +181,46 @@ class RequestImpl(HttpServerExchange exchange,
     }
 
     variable Map<String, String[]>? lazyQueryParameters = null;
-    value queryParameters => lazyQueryParameters 
+    value queryParameters => lazyQueryParameters
             else (lazyQueryParameters = readQueryParameters());
 
     variable FormData? lazyFormData = null;
-    value formData => lazyFormData 
+    value formData => lazyFormData
             else (lazyFormData = buildFormData()) ;
 
+    shared actual String[]? formParameters(String name) {
+        return formData.parameters[name];
+    }
+
+    shared actual String? formParameter(String name) {
+        if (nonempty x = formData.parameters[name]) {
+            return x.first;
+        }
+        else {
+            return null;
+        }
+    }
+
```

Comment:
"parameters" gets the query parameters from the Url and, (if there is
no such query parameter, or the flag is set, and the content type is
application/www-form-urlencoded), the form parameters from with in the
request body.

The query parameters (without the forms parameters) can be
parsed out of the result of the queryString method, so I did not add a
queryParam/queryParams method even though I would say that would be
better.

But, the form parameters can not be obtained alone. If you have a query
parameter and a form parameter and use the parameters method, you
cannot tell what was a query and what was a form parameter.

I'm implementing a REST server library using JAX-RS-like annotations
where you can specify if you want a query parameter or a form (or
header) parameter passed to your service implementation.

Hope this makes it clear, thanks,
Dirk